go/app/infra.go

```diff
@@ -35,7 +38,28 @@ func NewItemRepository() ItemRepository {
 
 // Insert inserts an item into the repository.
 func (i *itemRepository) Insert(ctx context.Context, item *Item) error {
-	// STEP 4-1: add an implementation to store an item
+	// STEP 4-2: add an implementation to store an item
+	filePath := "../go/items.json"
+	file, err := os.OpenFile(filePath, os.O_RDWR, 0644)
```

Comment:
[nits]
`i.fileName` で指定できます。この値を変えたい場合は、36行目の `	return &itemRepository{fileName: "items.json"}` の "item.json" を書き換えることで対応できます。

```suggestion
	file, err := os.OpenFile(i.fileName, os.O_RDWR, 0644)
```